src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -503,14 +503,27 @@ def aks_maintenanceconfiguration_add(
     config_name,
     config_file,
     weekday,
-    start_hour
+    start_hour,
```

Comment:
Oh, the problem about you cannot remove `required=False` comes from here. You'll need to set default value for these optional parameters, like `start_hour=None`.

Reply:
Can't thank you more! I was about to hit my head against the wall